enhancements/oc/top-pvc-usage.md

```diff
@@ -0,0 +1,117 @@
+---
+title: top-pvc-usage
+authors:
+- "@gmeghnag"
+- "@tsmetana"
+reviewers:
+- "@dobsonj"
+- "@ardaguclu"
+approvers:
+- "@ardaguclu"
+creation-date: 2024-10-23
+last-updated: 2024-10-23
+status:
+see-also:
+replaces:
+superseded-by:
+---
+
+# top pvc
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Provide a simple command to display a PersistentVolumeClaim capacity usage.
+
+## Motivation
+
+There already exist options like `oc adm top pod` that can display usage
+statistic of the server resources. Adding an option to display also usage
+of the PersistentVolumeClaims seems like a natural extension which provides
+important information about the persistent storage available to the workloads.
+
+The volume usage statistics are already available via the web console but we
+should also have a way to get this through the CLI too.
+
+### Goals
+
+1. Provide a simple CLI option to display filesystem usage of
+   PersistentVolumeClaims.
+2. Display only the percentual usage for a given PersistentVolumeClaim or all
+   PersistentVolumeClaims in a given namespace.
+
+### Non-Goals
+
+## Proposal
+
+Implement `oc adm top persistentvolumeclaims` command that would show usage
+statistic of the bound PersistentVolumeClaim like this:
+
+```text
+oc adm top pvc -n reproducer-pvc
+NAMESPACE      NAME               USAGE(%) 
+reproducer-pvc pvc-reproducer-pvc 98.28    
+reproducer-pvc pvc-test-pvc       14.56   
+```
+
+### User Stories
+
+#### Story 1
+
+As an OCP project user, I want to see a list of all PVC's in my namespace and
```

Comment:
- _"What is the expected permission set for this bearerToken that can successfully fetch the required metrics in here?"_
  At least the `ClusterRole/cluster-monitoring-view` and the `get/list` verbs on `Route` objects in `openshift-monitoring` namespace are needed to execute the query.
- _"Can a standard user use this command smoothly?"_
  A user without the above permissions can't use this command.
- _"or limited only to cluster admins?"_
  Being created under `oc adm` where most of the commands used for administrative tasks on the cluster reside, this command is firstly intended for cluster admins.
- _"If it is limited to cluster admins, wouldn't be less useful?, since we are providing a command to the users but only few of them can use it actually?"_
  I don't think so:
  - Being the cluster admins the people in charge of expanding volumes this command benefits them.
  - If needed, in a few steps, the following, anyone can execute such a command:
    ```
    $ oc create clusterrole routes-view --verb=get,list --resource=routes -n openshift-monitoring
    $ oc adm policy add-cluster-role-to-user routes-view <USER>
    $ oc adm policy add-cluster-role-to-user cluster-monitoring-view <USER>
    ```